### ISSUE TITLE
Sounds crack and pops using 48kHz and above frequencies with discreet audio cards

### ISSUE THREAD

**Issue by ericoporto** [adventuregamestudio/ags#1696]
**Describe the bug**
Explained by Dualnames [here](https://www.adventuregamestudio.co.uk/forums/index.php?topic=59842.msg636647051#msg636647051) and [here](https://www.adventuregamestudio.co.uk/forums/index.php?topic=59842.msg636647055#msg636647055) on the forums. Found with two different users, both with discreet audio cards, one specified. You can hear the issue [here](https://drive.google.com/file/d/18aGJbJisQdNvLuBzX6ujDfzNGzj7vPnE/view)!

The user has the sound device on Windows configured with 48kHz, and hear the crackling and popping. Setting the OS to use 44.1kHz fixes the sound.

> Windows 10, and his soundcard is "Soundblaster ZxR" and he's using "Voicemeeter Potatoe" for sound controls

Other user had a different configuration. 

I sent to Dualnames a build of only MojoAL test program `testposition.exe` and it reproduced the crackling/popping sound in the user machine - at 48kHz setting.

Because it sounded like https://github.com/libsdl-org/SDL/issues/5538, I built a version of SDL2 with libsamplerate (
[samplerate_sdl2.zip](https://github.com/adventuregamestudio/ags/files/8934384/samplerate_sdl2.zip)), and sent to Dualnames. The user reported this indeed fixed the audio.

**AGS Version**
It was AGS 3.6.0.24.

**Game**
The game it reproduced was Old Skies Demo, but theoretically any game with sound.

**To Reproduce**
Unfortunately it looks like it requires having a discreet audio card for this to happen. It **doesn't** reproduce in my Win10 machine.

**Expected behavior**
The sound should not have crackling or popping sound.

**Desktop:**
 - OS: Windows
 - Version 10

**Additional context**
This is possibly not an AGS issue, but I wanted to report here just to note down and separate from any other issue involving frequency. In theory we need to either wait [libsdl-org/SDL#5538](https://github.com/libsdl-org/SDL/issues/5538) or ship along libsamplerate on the Windows build.

**Comment by Dualnames1** [comment]
The other user that reported this issue said that their music is fixed with eri0o's patch, but not the voice/sounds.

**Comment by ivan-mogilko** [comment]
@Dualnames1 
> The other user that reported this issue said that their music is fixed with eri0o's patch, but not the voice/sounds.

could you summarize a difference between those, are they different in format, or anything else you may think of?

**Comment by Dualnames1** [comment]
all the files are oggs.

Their bitrate does vary.
There are audio files ranging from 32kbps to 323kbps, 90% of the audio files are at 44100 Hz, 6% at 48000 Hz
speech is at 58-170kbps,44100 Hz

I don't know why that is i can ask Dave to i guess go through all of them and match em. The difficulty here is the other user that is still experiencing this issue, is kinda difficult to get them to test anything. I don't have that issue myself, to my knowledge, i can ask how many people have downloaded the demo and finished it, but i think we're looking at a pool of like 1000 to 1.

**Comment by Dualnames1** [comment]
[2022-06-21_00-39-42.zip](https://github.com/adventuregamestudio/ags/files/8944576/2022-06-21_00-39-42.zip)

This is an external recording, the audio crackling starts at 0:32, i personally only hear it at 0:37-38 it's rather difficult to grasp it, but maybe im old af.

**Comment by ericoporto** [comment]
> This is an external recording, the audio crackling starts at 0:32, i personally only hear it at 0:37-38 it's rather difficult to grasp it, but maybe im old af.

I cannot hear anything wrong in the recording at all...

> I don't have that issue myself

From what I could understand, one of the issues this requires having a DAC or some other type of external sound card and it has to have unfortunate filters that makes the popping sounds audible, and this is made ok by using SDL2 built to work with libsamplerate, at least until the issue is fixed upstream.

The other issue you mention about voice/sounds is then a different issue, and this may not necessarily be correlated to the other.

**Comment by necros2k7** [comment]
Always wondered - isn`t 44,1khz enought for human ears (cd quality)?

**Comment by ivan-mogilko** [comment]
> We may need to start thinking about SDL3 and how/when we will migrate to it... (It's not released yet)

Is there a serious reason to do so, rather than fixing 1 bug? we just migrated to SDL2, it took roughly about a year to deal with various problems, and engine lost couple of functionalities in the progress that were not restored yet.

**Comment by ericoporto** [comment]
We can delay it and there's an adapter library to run SDL2 on top of SDL3. In terms of API most things are still there but they have a different naming scheme, so the call to things is instantly incompatible. We haven't added anything that uses an API that wasn't only renamed - I do manual check from time to time. The dollar touch (we don't use) has been moved to a satellite library. The is a new backend to the renderer called SDL_gpu that is an abstraction over 3D accelerated renderer with it's own shader language that compiles at runtime for all systems, and you can also access it and directly render stuff - this is the biggest change, and it's not merged in the main SDL3 branch yet, it's on icculus fork. Other than this internally mostly rewrites and refactors but with same functionality.
